pkg/query-service/app/http_handler.go

```diff
@@ -2262,6 +2275,130 @@ func (aH *APIHandler) logAggregate(w http.ResponseWriter, r *http.Request) {
 	aH.WriteJSON(w, r, res)
 }
 
+const logPipelines = "log_pipelines"
+
+func parseAgentConfigVersion(r *http.Request) (int, *model.ApiError) {
+	versionString := mux.Vars(r)["version"]
+
+	if versionString == "latest" {
+		return 0, nil
+	}
+
+	version64, err := strconv.ParseInt(versionString, 0, 8)
+
+	if err != nil {
+		return 0, model.BadRequestStr("invalid version number")
+	}
+
+	return int(version64), nil
+}
+
+func (ah *APIHandler) listLogsPipelinesHandler(w http.ResponseWriter, r *http.Request) {
+
+	version, err := parseAgentConfigVersion(r)
+	if err != nil {
+		RespondError(w, model.BadRequestStr("invalid version"), nil)
+		return
+	}
+
+	var payload *logparsingpipeline.PipelinesResponse
+	var apierr *model.ApiError
+
+	if version != 0 {
+		payload, apierr = ah.listLogsPipelinesByVersion(context.Background(), version)
+	} else {
+		payload, apierr = ah.listLogsPipelines(context.Background())
+	}
+
+	if apierr != nil {
+		RespondError(w, apierr, payload)
+		return
+	}
+	ah.Respond(w, payload)
+}
+
+// listLogsPipelines lists logs piplines for latest version
+func (ah *APIHandler) listLogsPipelines(ctx context.Context) (*logparsingpipeline.PipelinesResponse, *model.ApiError) {
+	// get lateset agent config
+	lastestConfig, err := agentConf.GetLatestVersion(ctx, logPipelines)
+	if err != nil {
+		if err != sql.ErrNoRows {
+			zap.S().Errorf("failed to get latest agent config version ", err)
```

Comment:
This will leave the ugly log message without a format specifier. If you use the `Error|Debug|Info(f)`, use the format specifiers or use the `Error` with zap fields. There are several places in the codebase where this is done including prior PRs. Please keep this detail in mind.

Reply:
Sure